pom.xml

```diff
@@ -50,6 +50,7 @@
         <module>components/org.wso2.carbon.identity.context.rewrite.valve</module>
         <module>components/org.wso2.carbon.identity.cors.service</module>
         <module>components/org.wso2.carbon.identity.cors.valve</module>
+        <module>components/org.wso2.carbon.identity.custom.header.filter</module>
```

Comment:
Are we not adding this as a feature to ship with product ?

Reply:
I don't think we should ship this by default with the product. In my opinion, it is better to keep it separate and add it whenever we need it. WDYT?

Reply:
Then we should move the config change to deployment.toml other than the web.xml.j2. We can add a way to define new filters from the deployment.toml

Reply:
Added in following PR. 
https://github.com/wso2/carbon-kernel/pull/2747